python/BioSimSpace/Sandpit/Exscientia/Align/_decouple.py

```diff
@@ -0,0 +1,95 @@
+import warnings
+
+from Sire import Base as _SireBase
+
+from .._SireWrappers import Molecule as _Molecule
+from .._Exceptions import IncompatibleError as _IncompatibleError
+
+
+def decouple(molecule, property_map0=None, property_map1=None, intramol=True):
+    """Make the molecule as being decoupled, where the interactions with the
+    rest of the environment are removed, or annihilate, where the interactions
+    within the molecule are removed as well (choose this mode with
+    intramol=False).
+
+        Parameters
+        ----------
+
+        molecule : BioSimSpace._SireWrappers.Molecule
+            The molecule to be decoupled or annihilated.
+
+        property_map0 : dict
```

Comment:
The use of `property_map` in this function is inconsistent with how it is used in the rest of the code. It should be treated as a dictionary that allows a user to remap the names of any properties, as desired. For example, a user could change the name of the _element_ property to _kitten_. If we then want to use the search functionality to find specific elements, then we need to pass this mapping to the search function, e.g.:

```python
system.search("element H", property_map={"element" : "kitten"})
```

In this case it appears that you are using the `property_map` as a means of specifying whether the charge or LJ terms are turned on or off in the two end states. If these are the only terms that you need to support, then it would probably be better to add two parameters to the function called `charge` and `LJ` (or `vdw`), which can take a two-element tuple specifying whether these terms are on or off in the end states. Defaults could be `charge=(True, False)`, `LJ=(True, False)`, which would match with what you currently have with the default property maps. (Alternatively you could add four parameters, e.g. `charge0`, `charge1`, `LJ0`, `LJ1`.) 

Also, it might also better to store the decoupling scheme in a different way, i.e. that doesn't conflict with the properties used for the exiting perturbable molecule. Perhaps you could store the decoupling scheme as another property, e.g. something like:

```python
edit_mol = edit_mol.setProperty("decouple", {"charge" : (True, False), "LJ" : (True, False)}).molecule()
```

You can then check to see if the molecule is decoupled by querying the property, then extracting the specific decoupling scheme, e.g.:

```python
if mol._sire_object.hasProperty("decouple"):
    decoupling = mol._sire_object.property("decouple")
    print(decoupling)
    print(decoupling["charge"])
    
Properties(
    LJ => [ 1,0 ],
    charge => [ 1,0 ]
)

[ 1,0 ]
```